include/openssl/ssl.h

```diff
@@ -1395,6 +1395,31 @@ __owur int SSL_CIPHER_get_bits(const SSL_CIPHER *c, int *alg_bits);
 __owur char *SSL_CIPHER_get_version(const SSL_CIPHER *c);
 __owur const char *SSL_CIPHER_get_name(const SSL_CIPHER *c);
 __owur uint32_t SSL_CIPHER_get_id(const SSL_CIPHER *c);
+typedef enum {
+    SSL_CIPHER_KX_RSA = 1,
+    SSL_CIPHER_KX_ECDHE,
+    SSL_CIPHER_KX_DHE,
+    SSL_CIPHER_KX_ECDHE_PSK,
+    SSL_CIPHER_KX_DHE_PSK,
+    SSL_CIPHER_KX_RSA_PSK,
+    SSL_CIPHER_KX_PSK,
+    SSL_CIPHER_KX_SRP,
+    SSL_CIPHER_KX_GOST
```

Comment:
As a side note, there are SSL_CIPHER_get_cipher_nid() and get_digest_nid(); I would've used NIDs if they were available, but they do not appear to be for these types.